src/main/java/org/springframework/data/repository/query/Parameters.java

```diff
@@ -62,11 +62,11 @@ public Parameters(Method method) {
 
 		Assert.notNull(method);
 
-		this.parameters = new ArrayList<T>();
-		this.dynamicProjectionIndex = -1;
-
 		List<Class<?>> types = Arrays.asList(method.getParameterTypes());
 
+		this.parameters = new ArrayList<T>(types.size());
```

Comment:
AFAICS moving the code shouldn't be necessary, should it? I usually declare variables as close as possible to their usage, so I'd like to keep the old position unless there's any effect of that move.